### DISCUSSION TITLE
能否实现改版前的自动下一集和自动连播功能分离？

### DISCUSSION THREAD

**Discussion by xgh827** [the1812/Bilibili-Evolved#1634] [功能建议 / Ideas]
现在开启自动连播后，相关推荐的视频也会被播放，而之前自动连播只对同一个视频的多个分P生效。请问能不能回退这个功能。
（在此问候决定这样改的**B站产品经理全家，这改动太蠢了）

**Comment by Coulomb-G** [comment]
自动连播和自动切集不都正常的吗?
这是两个功能且b站自带

**Reply by xgh827** [reply]
B站网页的新改动，自动切集和自动连播变成同一个功能了，要么都关要么都开。我百度了一下，有的人去年年底就接收到了这个改动，而我是这两天才变成这样。

**Comment by the1812** [comment]
截个图看看? 我至今没有往推荐视频播的功能, 自动连播放完所有分P就停了
![image](https://user-images.githubusercontent.com/26504152/110779195-36288180-829e-11eb-8d06-b58f7626f5e6.png)

**Reply by xgh827** [reply]
![image](https://user-images.githubusercontent.com/32972771/110780292-818f5f80-829f-11eb-9bd3-2fa72e778272.png)
在播放结束的这个页面，开启自动连播则左上角第一个推荐视频会开始转圈，转完自动跳到那个推荐的视频播放

**Reply by xgh827** [reply]
![image](https://user-images.githubusercontent.com/32972771/110780666-e6e35080-829f-11eb-8985-32e83f7278a4.png)
自动切集和自动连播这俩按钮是同步的，只能同时开或同时关

**Reply by the1812** [reply]
跟b站反馈了, 按理来说这两个应该是分开的

**Reply by Geeyun-JY3** [reply]
开隐身模式可以复现。
登录个人账号就和 repo 主一样，无分 p 视频「接下来播放」又变回「相关推荐」，自动连播开关也就跟着没了；分 p 视频下方的推荐视频标题行直接没了，就没有问题。

应该还在灰度，估计也没做完。再看看吧。

**Reply by Geeyun-JY3** [reply]
好的没想到这么快我也被灰度到了🌚

**Reply by the1812** [reply]
我今天也收到了, 可能是全量了

**Reply by xgh827** [reply]
> ![image](https://user-images.githubusercontent.com/32972771/110780292-818f5f80-829f-11eb-9bd3-2fa72e778272.png)
> 在播放结束的这个页面，开启自动连播则左上角第一个推荐视频会开始转圈，转完自动跳到那个推荐的视频播放

今天发现视频播放结束后向下滑动网页，则转圈会被自动取消

**Comment by RtYkk** [comment]
我也收到了，刚刚还准备去 issue 区开个 feature request ……还好仔细看了看 discussion，批站真是死了🐎，好的不学学些有的没的。

**Comment by Geeyun-JY3** [comment]
着急的可以试试这篇专栏文章的解决方案：https://www.bilibili.com/read/cv10762558

在脚本管理器导入以下代码（我这就是复制过来后把原文提醒要删除的 `?` 删掉了）
```
// ==UserScript==
// @name         bilibili ab test hook
// @namespace    http://tampermonkey.net/
// @version      0.1
// @description  try to take over the world!
// @author       小忍Alter
// @match        https://www.bilibili.com/video/*
// @run-at       document-body
// @grant        none
// ==/UserScript==

(function() {
    'use strict';

    // Your code here...
    window.webAbTest.play_continuously = false;

})();
```

实测有效。

（话说夜间模式是不是得适配一下专栏这个代码块

**Reply by the1812** [reply]
全量原因太草了
我目前也在考虑能不能去判断有多P/播放列表从而自动去开关那个自动播放 (虽然还是希望官方对此能有所作为

**Reply by Geeyun-JY3** [reply]
今天发现这个方法失效了，可以改用这个脚本：https://greasyfork.org/scripts/426452

**Comment by xgh827** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。
